lib/chef/client.rb

```diff
@@ -715,11 +718,57 @@ def self.generate_pfx_package(cert_name, date)
       pfx
     end
 
-    def create_new_key_and_register(cert_name)
+    def update_key_and_register(cert_name)
+      # Chef client and node objects exist on Chef Server already
+      # Create a new public/private keypair in secure storage
+      # and register the new public cert with Chef Server
       require "time" unless defined?(Time)
       autoload :URI, "uri"
 
-      # KeyMigration.instance.key_migrated = true
+      node = Chef::Config[:node_name]
+      d = Time.now
+      if d.month == 10 || d.month == 11 || d.month == 12
```

Comment:
@johnmccrae  Should we add some comment as to why these months should be checked and why the specific calculation for `end_date` under if-else block?

Reply:
resolved by redoing the date calcuation as marc suggested. Thanks for the feedback. This was a great idea.